pkg/crypto/ed25519.go

```diff
@@ -0,0 +1,111 @@
+package crypto
+
+import (
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"io/ioutil"
+	"os"
+
+	"golang.org/x/crypto/ed25519"
+)
+
+var (
+	// PubKeyIdentifier is the PEM public key identifier
+	PubKeyIdentifier = "PUBLIC KEY"
+	// PrivKeyIdentifier is the PEM private key identifier
+	PrivKeyIdentifier = "PRIVATE KEY"
+	// PEMCipher is the PEM encryption algorithm
+	PEMCipher = x509.PEMCipherAES256
+	// PubKeyFilePermissions are the public key file perms
+	PubKeyFilePermissions os.FileMode = 0644
+	// PrivKeyFilePermissions are the private key file perms
+	PrivKeyFilePermissions os.FileMode = 0600
+)
+
+// LoadPublicKeyFromFile loads PEM formatted ED25519 public key from file.
+func LoadPublicKeyFromFile(publicKeyPath string) ([]byte, error) {
+	x509PEM, err := ioutil.ReadFile(publicKeyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse x509 PEM file
+	var block *pem.Block
+	for {
+		block, _ = pem.Decode(x509PEM)
```

Comment:
this will try to decode x509PEM again and again, so if there is no PubKeyIdentifier as the first block it will loop forever. It should use the second value returned by `pem.Decode`, something like:
```
var (
    block *pem.Block
    remaining []byte = x509PEM
)
for {
    block, remaining = pem.Decode(remaining)
    if block == nil {
        return nil, errors.New("Cannot decode PEM file")
    }
    if block.Type == PubKeyIdentifier {
        break
    }
}
```

Reply:
Tried on the playground, https://play.golang.org/p/ZUgdLfB5Fvw